keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -9,27 +9,43 @@
   - [Non-Goals](#non-goals)
 - [Proposal](#proposal)
   - [Preserve existing behavior](#preserve-existing-behavior)
-  - [Updates to API server token generation](#updates-to-api-server-token-generation)
   - [New API](#new-api)
-  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  - [Plugins] (#plugins)
+  - [Updates to API server token generation](#updates-to-api-server-token-generation)
+  - [Updates for supported public keys](#updates-for-supported-public-keys)
+  - [Notes/Constraints/Caveats](#notesconstraintscaveats)
   - [Risks and Mitigations](#risks-and-mitigations)
-- [Graduation Criteria](#graduation-criteria)
+- [Design details](design-details)
+  - [KeyService RPC](#keyservice-rpc)
+  - [Test Plan](#test-plan)
+	  - [Prerequisite testing updates](#prerequisite-testing-updates)
+	  - [Unit tests](#unit-tests)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade/Downgrade Strategy](#upgradedowngrade-strategy)
+  - [Scalability](#scalability)
 - [Implementation History](#implementation-history)
 <!-- /toc -->
 
 ## Summary
 
-The Kubernetes API server has always read service account keys from disk as the process starts, and kept them in memory for the duration of the server's lifetime. As the API server can now verify and issue projected volume tokens, it would be advantageous to support external signing and verifying of token data over an API, as well as reading public keys from an API.
+ Kubernetes API server uses service account keys for signing JWT. Signing keys are stored on the disk and are loaded by Api server as the process starts. Key management is entirely within the cluster and is less flexible. It will be of advantage to provide a configurable capability that allows external JWT signing and key management.
 
 ## Motivation
 
-For operators who want to regularly rotate the signing and verifying keys for projected volume tokens, the Kubernetes API server must be restarted in order to use a new key. To facilitate easy key rotation, this KEP includes an proposal for a grpc API to support out of process signing and listing of signing keys.
+KEP proposes external JWT signing and listing of supported public keys.
+
+Benefits are 2 fold:
+1. Ease of rotation.
+  At present, ApiServer loads service account keys at process start time and the key remains the same for the life-time of the process. Thus, rotating keys require ApiServer restart. External key management eliminates the need to restart.
+2. Enhanced security.
+  ApiServer loads service account keys from a path on the disk. Which means that the service account keys are always present on the disk and can be easily accessed by the cluster administrators. With an external signer, we need not store the service account key on disk. 
 
 ### Goals
 
 - Support for out-of-process JWT signing
```

Comment:
We should explicitly call out that we want to make it possible to support HSMs and cloud KMSes.

Additionally, we should have as a goal that external signer implementations should not be able to introduce custom claims into the JWTs

Reply:
Done and Done